### ISSUE TITLE
Psychedelic glitch in Metroid Prime Remastered

### ISSUE THREAD

**Issue by XxAlcatelxX** [yuzu-emu/yuzu#10023]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Mainline 1390 / Early Access 3491

### Description of Issue

![image](https://user-images.githubusercontent.com/114865615/229311895-e5e64816-4145-4e4f-938d-de411406959d.png)

Hello, I hope you are well today, I would like to tell you about a bug that I suffered from the release of the Metroid Prime Remastered game, and I call it Rainbow Polygons or Psychedelic Glitch, I have tried to go through the settings all the time and the only solution to this glitch is to use OpenGL, but said API has another problem that was already mentioned in another github post, the best known RAM leak, the game eats all my ram to the point of crashing (happens after playing a few minutes or opening 2 doors). I suffered this problem in absolutely any version of Yuzu, either Early Access or Main

### Expected Behavior

![image](https://user-images.githubusercontent.com/114865615/229311879-d2aa7fcb-ac24-4a20-a9ed-2ef452c909a0.png)

### Reproduction Steps

To reproduce the problem, you must position yourself in a specific area of any map and everything will become Psychedelic or rainbow.

### Log File

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11130517/yuzu_log.txt)

### System Configuration

CPU: **Intel Core i7-4510U**
GPU/Driver: **Nvidia GeForce 840M 2GB VRAM**
RAM: **8 GB RAM DDR3 1600mhz**
OS: **Windows 11 64 bits**

**Comment by Ritaku02** [comment]
Question; do the glitched colors disappear when you move away from the spots that cause it, or do the colors stay glitched?

**Comment by XxAlcatelxX** [comment]
YES!

**Comment by XxAlcatelxX** [comment]
> Question; do the glitched colors disappear when you move away from the spots that cause it, or do the colors stay glitched?

Yes, do you also suffer from it? As I saw it was a problem with the Nvidia Maxwell graphics

**Comment by VladimirVSC** [comment]
I have the same problem. The glitch starts from the intro cut-scene and occurs periodically. The graphical driver is 470.161.03 (GTX 760)

[yuzu_log (Metroid prime Rainbow glitch).txt](https://github.com/yuzu-emu/yuzu/files/11158602/yuzu_log.Metroid.prime.Rainbow.glitch.txt) (if nesessary)

![Screenshot from 2023-04-05 15-13-02](https://user-images.githubusercontent.com/96921167/230078143-2581ed33-f2ab-4516-812d-d7f1ac171476.png)

![Screenshot from 2023-04-05 15-13-02](https://user-images.githubusercontent.com/96921167/230078223-aa66e4e9-eb2b-4385-a5b7-0df6d1ded9ae.png)

**Comment by XxAlcatelxX** [comment]
At least I'm not the only one suffering from this horrendous glitch, hopefully it can be resolved

**Comment by XxAlcatelxX** [comment]
Unfortunately it does not only happen in the introduction, in the whole game you will suffer the miserable glitch, I already made some progress and that's how it was, pure rainbow in my eyes, also the Decode ASTC Textures Asynchronously has problems in Metroid Prime Remastered, it is it takes too long to load textures unlike other games that require it, if you combine this option with this glitch, it will be all graphic chaos

**Comment by XxAlcatelxX** [comment]
> Pregunta; ¿Los colores defectuosos desaparecen cuando te alejas de los puntos que los causan, o los colores permanecen defectuosos?

Sorry, I forgot to answer your question correctly, and it is that when I move away from those points on the map that cause the glitch, the glitch disappears

**Comment by vasishath** [comment]
Wierdly, for me, this issue occurs only with OpenGL. And this appears immediately after an ASTC texture finishes loading, eg, a second after opening a door.

When I use vulkan, I never experience this. I think, this might have something to do with high stutter during ASTC decode as the stutter is quite high on opengl.

**Comment by XxAlcatelxX** [comment]
In my case, it stutters less in OpenGL than Vulkan, I heard that OpenGL was better for Nvidia users whose architecture is Fermi, Kepler and Maxwell. In my case, my graphics card is Maxwell (800 series)

**Comment by XxAlcatelxX** [comment]
> Extrañamente, para mí, este problema ocurre solo con OpenGL. Y esto aparece inmediatamente después de que una textura ASTC termine de cargarse, por ejemplo, un segundo después de abrir una puerta.
> 
> Cuando uso vulkan, nunca experimento esto. Creo que esto podría tener algo que ver con el tartamudeo alto durante la decodificación ASTC, ya que el tartamudeo es bastante alto en opengl.

I want to ask you a question, what hardware do you have? so to make sure which are the graphics that suffer from this glitch

**Comment by XxAlcatelxX** [comment]
> > > Extrañamente, para mí, este problema ocurre solo con OpenGL. Y esto aparece inmediatamente después de que una textura ASTC termine de cargarse, por ejemplo, un segundo después de abrir una puerta.
> > > Cuando uso vulkan, nunca experimento esto. Creo que esto podría tener algo que ver con el tartamudeo alto durante la decodificación ASTC, ya que el tartamudeo es bastante alto en opengl.
> > 
> > 
> > I want to ask you a question, what hardware do you have? so to make sure which are the graphics that suffer from this glitch, I mean, the psychedelic glitch and the stutters of the ASTC decoding
> 
> My hardware is Core i5-8250U NVIDIA MX150M (Pascal)

Pascal? How strange, I have seen users with graphics cards of the same architecture who were able to run Metroid Prime Remastered without bugs or glitches

**Comment by XxAlcatelxX** [comment]
Hello, there are new developments, Yuzu updated and added a new function called "Enable Reactive Flushing", this will help to eliminate the RAM leak, therefore, you will be able to play from OpenGL GLSL with almost no stutter and at the same time they will save the problem of the psychedelic Glitch, what's more, in my case I didn't even have to need "Decode ASTC Textures Asynchronously" for the game to go without stuttering, I am surprised by this fact, the psychedelic Glitch will still be presented in Vulkan, but for the most part less will be able to play it with OpenGL. 

I'm Happy😭

**Comment by vasishath** [comment]
> Hello, there are new developments, Yuzu updated and added a new function called "Enable Reactive Flushing", this will help to eliminate the RAM leak, therefore, you will be able to play from OpenGL GLSL with almost no stutter and at the same time they will save the problem of the psychedelic Glitch, what's more, in my case I didn't even have to need "Decode ASTC Textures Asynchronously" for the game to go without stuttering, I am surprised by this fact, the psychedelic Glitch will still be presented in Vulkan, but for the most part less will be able to play it with OpenGL.
> 
> I'm Happysob
> 
> Edit: Forget it, my game crashed after about 7 minutes, it just took longer to crash, the excitement got the better of mepensive

Reactive flushing will not fix the stutters. They are because of ASTC shaders decompressing which stall the GPU. For me, vulkan still has considerably less ASTC stutter than OpenGL.

**Comment by XxAlcatelxX** [comment]
Reactive Flushing adds stability in memory timing, that's why in my case the RAM leak took a bit longer to make Metroid Prime Remastered crash, as for stuttering, in my case it worked much better with OpenGL than with Vulkan , although the performance caused it to drop 20 FPS

**Comment by FunkyNico** [comment]
I can confirm i have this same issue playing with Vulkan and OpenGL, my hardware is:

Core i5-6300HQ
NVIDIA 940MX (Maxwell)

BTW i tried the suggested settings above with OpenGL and for what i have played i didn't notice any pyschedelic glitch but i haved a lot of stuttering in some scenarios, for example, at the begging where is two turrets and there's a room at the left where you can save the game and then is the big door where is the big parasit, every time i load the game i have a stutter in that zone and also a visual glich where you can see a few blank lines from the top like interlaced, but just a few moments and then goes normal, also personally i feel that Vulkan works better on my case, but there is the glitch present constantly, a lot also in that zone where is the big parasit that can shoot a laser from his mouth.

**Comment by Squall-Leonhart** [comment]
This is a duplicate of https://github.com/yuzu-emu/yuzu/issues/9146

**Comment by XxAlcatelxX** [comment]
> This is a duplicate of #9146

I wouldn't say that, my post came out before the one you mentioned but yes, from what I saw there is no solution, I will see if I send this problem to Nvidia

**Comment by Squall-Leonhart** [comment]
this issue was made this year, the one this is a duplicate of was made last year,

unfortunately its a problem i've been telling an nvidia contact about since 2019 when i first noticed that textures begin to scramble when gpu commit size exceeds gpu physical size considerably.

**Comment by vasishath** [comment]
I have a pascal GPU (MX 150) and for me, this bug is only reproduced on OpenGL, that too for like half a second or so. It has never produced on vulkan and vulkan also gives much lesser delay and stutter on opening a door.

**Comment by XxAlcatelxX** [comment]
Now that you mention it, I'll try increasing the driver cache in the Nvidia control panel and using a little more virtual memory to see what happens.

**Comment by XxAlcatelxX** [comment]
It doesn't work, I was hoping it won't work...![IMG_20231007_095729.jpg](https://github.com/yuzu-emu/yuzu/assets/114865615/1903eac3-b36f-4392-9a85-28e0f8e815e8)
